src/components/CarouselContainer.ts

```diff
@@ -120,21 +122,25 @@ export default class CarouselContainer extends Component<CarouselContainerProps,
             });
             this.setState({ images, isLoading: false });
         } else if (this.props.dataSource === "XPath" && this.props.imagesEntity) {
-            this.fetchImagesByXPath(mxObject ? mxObject.getGuid() : "");
+            if (mxObject) {
+                this.fetchImagesByXPath(mxObject);
+            }
         } else if (this.props.dataSource === "microflow" && this.props.dataSourceMicroflow) {
             this.fetchImagesByMicroflow(this.props.dataSourceMicroflow, mxObject);
         }
     }
 
-    private fetchImagesByXPath(contextGuid: string) {
+    private fetchImagesByXPath(mxObject: mendix.lib.MxObject) {
         const { entityConstraint } = this.props;
         const requiresContext = entityConstraint && entityConstraint.indexOf("[%CurrentObject%]") > -1;
+        const contextGuid = mxObject.getGuid();
         if (!contextGuid && requiresContext) {
             this.setState({ images: [], isLoading: false });
             return;
         }
 
         const constraint = entityConstraint ? entityConstraint.replace("[%CurrentObject%]", contextGuid) : "";
```

Comment:
do recursive replacement of `[%CurrentObject%]` by using regular expressions.
`enityConstraint.replace(/\[%CurrentObject%\]/g, contextGuid)`

Reply:
Done